invertedai/api/validations.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Optional, Dict
+from invertedai.error import InvalidInputType, InvalidInput
+from invertedai.common import (
+    RecurrentState,
+    AgentState,
+    AgentAttributes,
+    TrafficLightId,
+    TrafficLightState,
+)
+
+
+def drive_validation(drive):
+    def wrapped_drive(
+        location: str,
+        agent_states: List[AgentState],
+        agent_attributes: List[AgentAttributes],
+        recurrent_states: List[RecurrentState],
+        traffic_lights_states: Optional[Dict[TrafficLightId, TrafficLightState]] = None,
+        get_birdview: bool = False,
+        get_infractions: bool = False,
+        random_seed: Optional[int] = None,
+    ):
+        try:
+            assert isinstance(location, str)
+            assert all(map(lambda s: isinstance(s, AgentState), agent_states))
+            assert all(map(lambda s: isinstance(s, AgentAttributes), agent_attributes))
+            assert all(map(lambda s: isinstance(s, RecurrentState), recurrent_states))
+            assert isinstance(traffic_lights_states, (list, type(None)))
+            assert isinstance(get_birdview, bool)
+            assert isinstance(get_infractions, bool)
+            assert isinstance(random_seed, (int, type(None)))
```

Comment:
It's generally desirable to allow for duck typing in Python, accepting objects that have different types but the same interface. As a rule of thumb, if the rest of the function was going to execute correctly with some input and return a correct result, this input should not be rejected.